webdriver-spec.html

```diff
@@ -5213,18 +5224,85 @@ <h3 id="character-types">Character types</h3>
 </section> <!-- /Modals -->
 
 <section>
-<h2>Screenshots</h2>
+<h2>Screen Capture</h2>
+
+<p>Screenshots are a mechanism for providing
+ additional visual diagnostic information.
+ They work by dumping a snapshot of
+ the <a href=https://drafts.csswg.org/css-device-adapt/#the-viewport>viewport</a>’s
+ framebuffer as a lossless PNG image.
+ It is returned to the <a>local end</a> as a Base64 encoded string.
+
+<p>WebDriver provides the <a>Take Screenshot</a> <a>command</a>
+ to capture the
+ <a href=https://html.spec.whatwg.org/#top-level-browsing-context>top-level browsing context</a>’s
+ <a href=https://drafts.csswg.org/css-device-adapt/#the-viewport>viewport</a>,
+ and a <a>command</a> <a>Take Element Screenshot</a>
+ for doing the same with the visible region of
+ an <a href=https://dom.spec.whagwg.org/#concept-element>element</a>’s
+ bounding rectangle after it has been <a>scrolled into view</a>.
+
+<p>In order to <dfn>draw a bounding box from the framebuffer</dfn>, given
+ a <a href=https://drafts.fxtf.org/geometry/#rectangle>rectangle</a>
+ <var>rect</var>:
 
-<p class=issue><strong>Warning:</strong>
- This section has not yet been redefined to match the <a>routing requests</a> model,
- and uses old concepts and definitions.
- Please do not rely on it yet.
+<ol>
+ <li><p>Let <var>viewport width</var> and <var>viewport height</var>
+  be the size of the
+  <a href=https://drafts.csswg.org/css-device-adapt/#initial-viewport>initial viewport</a>.
+
+ <li><p>If either <var>viewport width</var>
+  or <var>viewport height</var> is &lt; 0,
+  return <a>error</a> with <a>error code</a>
+  <a>unable to capture screen</a>.
+
+ <li><p>Let <var>paint width</var> be
+  <var>viewport width</var> – <var>rect</var>’s
+  <a href=https://drafts.fxtf.org/geometry/#rectangle-width-dimension>width dimension</a>.
+
+ <li><p>Let <var>paint height</var> be
+  <var>viewport height</var> – <var>rect</var>’s
+  <a href=https://drafts.fxtf.org/geometry/#rectangle-height-dimension>height dimension</a>.
+
+ <li><p class=issue>Determine if all corners of <var>rect</var> is
+  inside the rectangle that is the viewport.
+
+ <li><p>Let <var>canvas</var> be a new
```

Comment:
Added that prose.

The point of the algorithms is to give clarity to what the result or output of a command would look like, however.  Not necessarily to define exactly how a user agent might implement something.